src/nickel.rs

```diff
@@ -156,7 +157,15 @@ impl<D: Sync + Send + 'static> Nickel<D> {
         });
 
         let server = Server::new(self.middleware_stack, self.data);
-        let listener = server.serve(addr, self.keep_alive_timeout).unwrap();
+
+        let is_test_harness = env::vars().any(|(ref k, _)| k == "NICKEL_TEST_HARNESS");
```

Comment:
Can you clarify why the examples can't just keep running on one defined port? I'm assuming it's because it would prevent us from running multiple tests in parallel but as far as I understood it so far that won't be the case anyway because cargo doesn't work well in parallel either.